### DISCUSSION TITLE
Prevent error relating to deprecation of ActiveSupport::Logger.broadcast in Rails 7.1

### DISCUSSION THREAD

**Discussion by kramerdog** [sidekiq/sidekiq#6164] [Q&A]
I recently tried to upgrade to Rails 7.1 but this introduced a Sidekiq error due to the deprecation of ActiveSupport::Logger.broadcast

I understand that this was fixed in this commit https://github.com/sidekiq/sidekiq/commit/a1a220f7939fc422c89706454b3bd6799dcf47d2#diff-d68404a76acb05ee767e7631f9e9acb7e9c5db646aaae1e2c772fe371fae1e85 
however it seems to only apply to Sidekig 7+

I believe that I cant upgrade my app to Sidekiq 7 because there is a dependency on Redis 6 and there doesnt seem to be a Gem released yet for Redis 6.

So I need to find a solution for Sidekiq 6.5 - any suggestions?

**Comment by mperham** [comment]
I don't plan on adding this. Sidekiq 6.5 supports Rails 7.0 but if you want Rails 7.1, you should be upgrading Sidekiq too.

If you give us more info on the gem incompatibility, maybe we can find a workaround or open an issue to get it fixed.

**Comment by kramerdog** [comment]
Hi Mike

Thanks very much for the prompt response.

If I comment out the version 6.5 constraint for Sidekiq in my Gemfile

gem 'sidekiq'#, '~> 6.5'

and bundle update so that I have:

sidekiq (7.2.0)
  concurrent-ruby (< 2)
  connection_pool (>= 2.3.0)
  rack (>= 2.2.4)
  redis-client (>= 0.14.0)

and

redis (5.0.8)
  redis-client (>= 0.17.0)
redis-client (0.19.1)
  connection_pool


in Gemfile.lock

Then when I start up Sidekiq I get:

11:02:53 sidekiq.1 | 2024-01-15T01:02:53.159Z pid=525774 tid=bd6u INFO:
Sidekiq 7.2.0 connecting to Redis with options {:size=>10,
:pool_name=>"internal", :url=>nil}
11:02:53 sidekiq.1 | redis-client requires Redis 6+ with HELLO command
available (redis://localhost:6379/0)
11:02:53 sidekiq.1 |
***@***.***_ancestors/gems/redis-client-0.19.1/lib/redis_client.rb:759:in
`rescue in connect'

and Sidekiq terminates.

However there doesnt appear to be a Redis 6 gem available to fulfil
*redis-client
requires Redis 6+ with HELLO command available*

Maybe I am missing something?

**Reply by mperham** [reply]
Redis the server, not the gem.

**Comment by mperham** [comment]
Please read the 7.0 upgrade notes in docs/ if you haven’t already.

**Comment by kramerdog** [comment]
Thanks Mike

I appreciate the responses.

I can see the notes in the upgrade doc
redis-client

The redis-client gem is a new Rubygem which uses the RESP3 protocol found
in Redis 6.0+. Sidekiq 6.5 introduced beta support for the redis-client gem
while defaulting to using the redis gem. Sidekiq 7.0 completes this
transition and no longer uses the redis gem internally. *Your app can still
continue to use redis.* If you use Sidekiq.redis to access Redis
connections, that API is now exposing a connection based on redis-client.

I am afraid in spite of this it is still not clear to me what I need to do
to avoid the error

11:02:53 sidekiq.1 | redis-client requires Redis 6+ with HELLO command
available (redis://localhost:6379/0)

How do I upgrade to Redis 6+ if the latest Redis gem available is 5.0.8?

gem search '^redis$' --remote

*** REMOTE GEMS ***

redis (5.0.8)

Apologies if I am missing something obvious here?

Regards

Craig

**Reply by raybaxter** [reply]
You need to distinguish between the [Redis server](https://redis.io/) which is an in-memory value store, and the [`redis` gem](https://github.com/redis/redis-rb) which is a gem providing an interface for connecting to a Redis server. 

This is similar to MySQL and the `mysql2` gem, or PostgreSQL and the `ruby-pg` gem.

Redis server version 6.2 (and 7.0) is available for [download](https://redis.io/download/) if you are hosting locally, and is also available in various images, from your hosting provider, etc. 

As you've noted, the latest version of the `redis` gem is 5.0.8.

**Reply by kramerdog** [reply]
Thanks for the response Ray.

Yes, I have been a bit confused obviously. I understand now that you are saying that I need to update my Redis server installation to version 6+.

Cheers.
